themes/src/main/resources/theme/base/login/register.ftl

```diff
@@ -136,11 +129,13 @@
             <@registerCommons.termsAcceptance/>
 
             <#if recaptchaRequired??>
-                <div class="form-group">
-                    <div class="${properties.kcInputWrapperClass!}">
-                        <div class="g-recaptcha" data-size="compact" data-sitekey="${recaptchaSiteKey}"></div>
+                <#if recaptchaVisible!false>
```

Comment:
Since this condition does not have another branch, I think we can roll this into the `recaptchaRequired` check, so there is only a single `if` statement checking both conditions.